### PR TITLE
feat: Implement print intent

### PR DESCRIPTION
**Pull request by acezard**
- [x] AA - Interface permettant au cozy-app de demander une impression (cozy-intent.call(print) - la cozy-app fourni la base 64 du fichier à imprimer )
- [x] AA - Intégrer la librairie  https://github.com/christopherdro/react-native-print
- [x] AA - Sauvegarde du fichier dans un dossier temporaire (voir OCR)
- [x] AA - Afficher un toast si erreur
- [x] AA - Avoir une méthode de type (isprintavailable)
- [x] Add tests

### CODE REVIEW

File: src/libs/intents/printBase64Doc.ts [hunk 1]
```diff
@@ -11,56 +11,66 @@ import { t } from '/locales/i18n'
...
     if (!base64) throw new Error('No base64 provided')
+    path = getTemporaryPrintFilePath(getFileTypeFromDataURI(base64))
+    const fileContent = getContentFromBase64(base64)
 
     await ensurePrintFolder()
-    await RNFS.writeFile(path, base64, 'base64')
-
-    await RNPrint.print({ filePath: path })
+    await RNFS.writeFile(path, fileContent, 'base64')
+    await RNPrint.print({ filePath: path.replace('file://', '') })
```
Comment (zatteo) [inline src/libs/intents/printBase64Doc.ts:71]: Why do you need to do a `path.replace` again ? It is already in `getTemporaryPrintFilePath` no ?

Comment (acezard) [reply src/libs/intents/printBase64Doc.ts:71]: Yes, need to remove it. With the prefix the print lib can't find the file. Maybe writeFile could not use it so it's clearer

Comment (acezard) [reply src/libs/intents/printBase64Doc.ts]: @zatteo to sum up: from what I understand I need the file prefix when writing the file, but when reading it with the print library, having that prefix will make the print software unable to read the file. I made it clearer by abstracting this file prefix in a "writeFile" local function

Comment (zatteo) [reply src/libs/intents/printBase64Doc.ts]: For backup feature, it seems that I remove 'file://' both for iOS and Android.

I do not know what it is kept on o[cr feature / if it's needed here](https://github.com/cozy/cozy-flagship-app/blob/master/src/app/domain/ocr/services/ocr.ts#L61).

File: src/libs/intents/printBase64Doc.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    if (await RNFS.exists(path)) {
+      await RNFS.unlink(path)
+    }
+  } catch (err) {
+    log.error(`Error deleting document: ${getErrorMessage(err)}`)
+  }
+}
+
+const writeFile = async (filePath: string, content: string): Promise<void> => {
+  return await RNFS.writeFile(`${filePath}`, content, 'base64')
```
Comment (Ldoppea) [inline src/libs/intents/printBase64Doc.ts:40]: nit: we don't need this templated string anymore here.
```suggestion
  return await RNFS.writeFile(filePath, content, 'base64')
```

File: src/libs/intents/printBase64Doc.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    filePath = getTemporaryPrintFilePath(getFileTypeFromDataURI(base64))
+    const fileContent = getContentFromBase64(base64)
+    await ensurePrintFolder()
+    await writeFile(filePath, fileContent)
+    await RNPrint.print({ filePath })
+  } catch (err) {
+    log.error(`Error while printing document: ${getErrorMessage(err)}`)
+    Toast.show({
+      type: 'error',
+      text1: t('error.unknown_error')
```
Comment (Ldoppea) [inline src/libs/intents/printBase64Doc.ts:77]: Should we be more specific in the toast message to state that this is about the print feature? Something like `Failed to print, an error occured`

**Review by Ldoppea — APPROVED** [review]
Nice PR, commits are well split so it was easy to read 👍

File: src/libs/intents/printBase64Doc.ts [hunk 4]
```diff
@@ -11,56 +11,66 @@ import { t } from '/locales/i18n'
...
 const getPrintFolderPath = (): string => {
   return `${RNFS.TemporaryDirectoryPath}/print/`
 }
 
-const getTemporaryPrintFilePath = (): string => {
+const getTemporaryPrintFilePath = (fileExtension: string): string => {
   const date = format(new Date(), 'yyyyMMddHHmmssSSS')
-  const tempFileName = `temp_print_${date}.printfile`
-
+  const tempFileName = `temp_print_${date}.${fileExtension}`
```
[the remark this answers is not in the document]
Comment (acezard) [reply src/libs/intents/printBase64Doc.ts:20]: I didn't see it, and when saving as PDF Android discards the temp file name and creates its own filename from scratch. Though in any case, I make the name more end-user presentable if it's ever shown

File: src/libs/intents/printBase64Doc.ts (hunk 2 above)
Comment (acezard) [reply src/libs/intents/printBase64Doc.ts:40]: ✅

File: src/libs/intents/printBase64Doc.ts (hunk 3 above)
Comment (acezard) [reply src/libs/intents/printBase64Doc.ts:77]: agree ✅

**Pull request merged**